python/dgl/convert.py

```diff
@@ -501,8 +501,10 @@ def heterograph(data_dict, num_nodes_dict=None, restrict_format='auto',
         num_nodes_dict = defaultdict(int)
         for (srctype, etype, dsttype), data in data_dict.items():
             if isinstance(data, tuple):
-                src = utils.toindex(data[0]).tonumpy()
-                dst = utils.toindex(data[1]).tonumpy()
+                src = utils.toindex(data[0],
+                                    dtype='int32' if idtype == F.int32 else 'int64').tonumpy()
+                dst = utils.toindex(data[1],
+                                    dtype='int32' if idtype == F.int32 else 'int64').tonumpy()
```

Comment:
Add a TODO here:
`TODO(minjie): converting to index is unnecessary just for getting the max ID.`
I will change them later when I get rid of all the Index object.

Reply:
Done